### PR TITLE
Add Monospace Font Support to the Editor

### PR DESCRIPTION
**Pull request by roundhill**
This PR adds a new setting to allow display of a monospace font in the note editor.

Windows:
screen shot 2018-01-02 at 2 53 46 pm

Linux:
screen shot 2018-01-02 at 2 40 25 pm

New menu item:
screen shot 2018-01-02 at 2 44 34 pm

**To Test**
 * Enable the monospace font, either in the 'View' menu or in the app settings dialog.
 * The editor should switch to the monospace font, and back to the default system font if disabled.

@drw158 'Courier' seemed to be the best looking cross-platform monospace font, but would love any additional input you may have.

### CODE REVIEW

**Comment by drw158** [comment]
Might be cool to use a single open source monospace font, instead of relying on the system's fonts. But, we do already use the default system font for sans serif.

Here is a font-stack I pulled from Medium

```font-family: Menlo, Monaco, "Courier New", Courier, monospace;```

and one from Github:

```"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace;```

I don't understand the Medium one, since most of those are Apple-only I think. I like the Github one, although I haven't come across SFMono before — would be cool if it worked for us and the Electron Mac version. Courier strays pretty far from what a "normal" monospace font looks like nowadays, so I like that they put it last.

**Comment by drw158** [comment]
Just came across this article from IA Writer:

https://ia.net/topics/in-search-of-the-perfect-writing-font/

> If you don’t own a copy of our writing app, you will find iA Writer Duospace on [GitHub](https://github.com/iaolo/iA-Fonts/tree/master/iA%20Writer%20Duospace), so you can use it in the writing app of your choice. We are looking forward to hearing how it feels to you.

Looks like they put in a monumental amount of effort to choose the perfect writing font. If we want to bundle a font in SN, this might be a good monospace font option.

**Comment by roundhill** [comment]
> Looks like they put in a monumental amount of effort to choose the perfect writing font.

Nice, let's give it a try.

**Comment by roundhill** [comment]
Here's a screen with `iA Writer Duospace`. Looks nice!

screen shot 2018-01-09 at 2 45 48 pm

We will need to include another license file to use it, but it seems compatible with our current license.

**Comment by drw158** [comment]
It looks great! We will need to check language support. 

If that font doesn't work, there are a lot of other open source fonts suggested in the original issue #274

**Comment by roundhill** [comment]
> It looks great! We will need to check language support.

Do you know of a good way to test that?

**Comment by drw158** [comment]
Sorry for the slow response... I'm not sure, I just remember some language issues people had with Source Sans. My best guess is to create a user on your computer and switch the language?

File: lib/app.jsx [hunk 1]
```diff
@@ -89,6 +89,7 @@ function mapDispatchToProps(dispatch, { noteBucket }) {
     setAuthorized: () => dispatch(reduxActions.auth.setAuthorized()),
     setSearchFocus: () =>
       dispatch(actionCreators.setSearchFocus({ searchFocus: true })),
+    toggleMonospace: () => dispatch(settingsActions.toggleMonospace()),
```
Comment (rodrigoi) [inline lib/app.jsx]: you can update skip this and add `toggleMonospace` to the `pick(settingsActions)` array.

**Comment by rodrigoi** [comment]
@roundhill I just gave this a try and works fine. Looks like it need a rebase and should be ready to go!
Let me know if you need more help with this.

**Comment by webmasterraj** [comment]
It would be awesome to get this merged in. Anything I can do to help?

**Comment by dmsnell** [comment]
@roundhill @rodrigoi I have rebased this against the current master

@webmasterraj thanks for the offer! always helpful is running the branch through some testing and even more with font issues to run some tests with non-latin characters.

it seems like this may have been ready to go but got a little lost in the fray; let's see what @roundhill says…

**Comment by yoyoyojomo** [comment]
I am just randomly stumbling upon this PR, but I thought I should mention that Duospace is not a fixed-width font; 'm' and 'w' are 1.5 width. This could be confusing for anyone expecting fixed-width like ASCII tables/art/etc.

**Comment by roundhill** [comment]
We're still not quite there with deciding on different font support across the apps, going to close this for now but we can keep the branch for the future once we're ready to go.

**Comment by pervoj** [comment]
What does it look like with a monospace font? That would be really great.

**Comment by codebykat** [comment]
@pervoj This is a very old PR, but you might be interested in the discussion on this issue: https://github.com/Automattic/simplenote-electron/issues/274#issuecomment-892934934

**Pull request closed without merging**